rosbag2_cpp/src/rosbag2_cpp/reindexer.cpp

```diff
@@ -188,6 +190,13 @@ void Reindexer::aggregate_metadata(
     auto temp_metadata = bag_reader->get_metadata();
     metadata_.storage_identifier = temp_metadata.storage_identifier;
 
+    // try to find the last log for the most complete custom data section
+    if (latest_log_start < temp_metadata.starting_time) {
+      latest_log_start = temp_metadata.starting_time;
+      metadata_.custom_data = temp_metadata.custom_data;
+      metadata_.ros_distro = temp_metadata.ros_distro;
+    }
+
```

Comment:
a quick question. are we expecting that `custom_data` and `ros_distro` are different from file to file? if that is not time sensitive information and static, we can store once for each element?

Reply:
If update_metadata is used on the underlying storage, both of these can change between successive split files. 

Personally, I would argue that `ros_distro` is probably far less likely to be subject to change. However, I currently have a custom writer implementation that supports updating the `custom_data` field during recording, meaning that the data is changed progressively between split files. I use this for indicating events that the log contains as well as other custom data I want to store with the log while the system is being recorded.

The only reason for `ros_distro` field being updated to the value from the youngest split file is simply because I had already added the logic to find the youngest file for `custom_data`.